src/Magnum/Shaders/Generic.h

```diff
@@ -358,7 +358,26 @@ template<UnsignedInt dimensions> struct Generic {
      */
     typedef GL::Attribute<5, Vector3> Normal;
 
-    /* 6, 7 reserved for vertex weights / bone IDs */
+    /**
+     * @brief Skin weights
+     *
+     * @ref Magnum::Vector4 "Vector4", four weights of influence per @ref JointIndices.
+     * Corresponds to @ref Trade::MeshAttribute::Weights.
+     */
+    // TODO: E.g. gltf supports importing multiple sets of 4 weights (WEIGHTS_0 [0;3], WEIGHTS_1 [4;7])
+    // I suppose these would require special shaders and it's fine to only support 4 for the GenericShaders
```

Comment:
Yep, and that's why I reserved the "2nd vertex weights" etc above. This would be then a `SecondaryWeights` attribute, and yes the shader needs to explicitly support that (it's not fine to hard-cap the support to just 4, no :P).

Hmm, now i'm thinking if it's actually possible for the shader to access both primary and secondary weights through the same accessor (maybe through a `mat2x4` where the first column is bound to id 6 and the second to id 10?)

Reply:
Why would it need to access both weight sets through the same accessor? The Gltf 2.0 overview seems to use a very "unrolled" way of multiplying weights with bone transforms, no matrix multiplications there:

```glsl

uniform mat4 u_jointMatrix[12];
in vec4 position;
in vec4 jointIds;
in vec4 weights;

void main(void) {
    mat4 skinMatrix =
        weights.x*jointMatrix[int(jointIds.x)] +
        weights.y*jointMatrix[int(jointIds.x)] +
        weights.z*jointMatrix[int(jointIds.x)] +
        weights.w*jointMatrix[int(jointIds.x)];

    gl_Position = modelViewProjection*skinMatrix*position;
}
```

Reply:
Shouldn't that be `jointIds.xyzw` also? I mean the case with

```glsl
in vec4 jointIds;
in vec4 secondaryJointIds;
in vec4 weights;
in vec4 secondaryWeights;
```

is kinda ugly, so I'm wondering if this could be turned into some

```glsl
in float jointIds[8];
in float weights[8];
```

instead, with the `8` being possibly passed in during compilation (so e.g. we can support 5 influences without always doing calculations for 8.

Reply:
> Shouldn't that be jointIds.xyzw also?

Yes! => fixed

> pseudocode

This will lead to some ugly macro stuff if you do it this way. Instead with the duplicate attributes, it'd be just a block like `#ifdef MORE_BONES` which would be added rather than replacing the code that works for 4 joints/2 attributes.

Reply:
> ugly macro

why? this is what I wanted to achieve

```glsl
mat4 skinMatrix;
for(int i = 0; i != HOW_MANY_THEM_BONES; ++i)
    skinMatrix += weights[i]*jointMatrices[int(jointIds[i])];
```

Reply:
Wouldn't `mat2x4 jointIds` return a vec4 on `jointIds[0]` though?

Reply:
This is becoming endless :sweat_smile: 

Yes it would. I'm referring to the `in float jointIds[8];` above.